airbyte-integrations/connectors/source-shopify/source_shopify/shopify_graphql/graphql.py

```diff
@@ -0,0 +1,471 @@
+
+        def record_resolve_id(record: Mapping[str, Any]) -> Mapping[str, Any]:
+            """
+            The ids are fetched in the format of: " gid://shopify/Order/<Id> "
+            Input:
+                { "Id": "gid://shopify/Order/19435458986123"}
+            We need to extract the actual id from the string instead.
+            Output:
+                { "id": 19435458986123, "admin_graphql_api_id": "gid://shopify/Order/19435458986123"}
+            """
+            # save the actual api id to the `admin_graphql_api_id`
+            record["admin_graphql_api_id"] = record["id"]
+            # extracting the int(id) and reassign
+            record["id"] = int(re.search(r"\d+", record.get("id")).group())
+            return record
+
+        def resolve_substream(record: Mapping[str, Any], record_identifier: str = None) -> Mapping[str, Any]:
+            # return records based on `record_identifier` filter.
+            if record_identifier:
+                return record if record_identifier in record.get("admin_graphql_api_id") else None
+            else:
+                # return records related to substream, by checking for the `__parentId` field
+                # more info: https://shopify.dev/docs/api/usage/bulk-operations/queries#the-jsonl-data-format
+                return record if ShopifyBulkGraphQl.parent_key in record.keys() else None
+
+        def resolve_with_custom_transformer(record: Mapping[str, Any], custom_transform: Callable) -> Iterable[Mapping[str, Any]]:
+            if isgeneratorfunction(custom_transform):
+                yield from custom_transform(record)
+            else:
+                yield custom_transform(record)
+
+        def record_resolver(
+            record: Mapping[str, Any],
+            substream: bool = False,
+            record_identifier: str = None,
+            custom_transform: Callable = None,
+        ) -> Iterable[Mapping[str, Any]]:
+            # resolve the id to int
+            record = ShopifyBulkGraphQl.Tools.record_resolve_id(record)
+            # the substream_record is `None`, when parent record takes place
+            substream_record = ShopifyBulkGraphQl.Tools.resolve_substream(record, record_identifier)
+
+            # resolution
+            if custom_transform and substream:
+                # process substream with external function, if passed
+                if substream_record:
+                    yield from ShopifyBulkGraphQl.Tools.resolve_with_custom_transformer(substream_record, custom_transform)
+            elif custom_transform and not substream:
+                # process record with external function, if passed
+                yield from ShopifyBulkGraphQl.Tools.resolve_with_custom_transformer(record, custom_transform)
+            elif not custom_transform and substream:
+                # yield substream record as is
+                if substream_record:
```

Comment:
nothing will be yielded if this condition is false. is this intentional?

Reply:
Same as above, but for the scenario with `custom_transform` involved.